### PR TITLE
Initial text-to-speech (TTS) docs

### PR DESCRIPTION
**Pull request by fabaff**
**Description:**
Initial text-to-speak (TTS) docs

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#4837

### CODE REVIEW

File: source/_components/tts.google.markdown [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Example configuration.yaml entry
+tts:
+  - platform: google
+```
+
+Configuration variables:
+
+- **language** (*Optional*): The language to use. Defaults to `en`.
+- **cache** (*Optional*): If you want to use the caching feature. Defaults to `true`.
+- **cache_dir** (*Optional*): The port on which the APCUPSd NIS is listening. Defaults to `tts`.
```
Comment (robbiet480) [inline source/_components/tts.google.markdown]: Stale description

**Comment by robbiet480** [comment]
It should be text-to-speech, not text-to-speak.

**Comment by fabaff** [comment]
Thanks

File: source/_components/tts.google.markdown [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+layout: page
+title: "Google"
```
Comment (mnoorenberghe) [inline source/_components/tts.google.markdown]: This title should be less generic e.g. "Google Text-to-Speech"

File: source/_components/tts.google.markdown [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: google.png
+ha_category: Text-to-speech
+ha_release: 0.35
+---
+
+The `google` text-to-speech platform uses [Google](https://www.google.com) Text-to-Speech engine to read a text with natural sounding voices.
```
Comment (mnoorenberghe) [inline source/_components/tts.google.markdown]: Can we link "Google Text-to-Speech engine" to Google API docs?

File: source/_components/tts.markdown [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+sharing: true
+footer: true
+ha_release: 0.35
+---
+
+Text-to-speech (TTS) enables Home Assistant to speak to you.
+
+# {% linkable_title Cache %}
+
+The component have 2 cache. A long time cache on file system, they can controll with `cache` option on platform config or service call `say`. And a in memory cache for fast response to media player, they will be auto cleanup after several time.
```
Comment (mnoorenberghe) [inline source/_components/tts.markdown]: This paragraph needs rephrasing and typo fixes.

File: source/_components/tts.markdown [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+To get started add the following lines to your `configuration.yaml` (example for google):
+
+```yaml
+# Example configuration.yaml entry for google tts service
+tts:
+  - platform: google
+```
+
+The following optional parameters can be used with any platform. However tts component will only look for global settings under the configuration of the first configured platform:
```
Comment (mnoorenberghe) [inline source/_components/tts.markdown]: Note about the implementation, not docs: Why are we adding this limitation to a new platform? It's something I understand is supposed to be fixed in other platforms.

Comment (fabaff) [inline source/_components/tts.markdown]: I guess that @pvizeli could answer this question be than me. From my point of view it looks like that we don't want to handle multiple caches.

**Pull request merged**